Src/PerformanceCollector/Shared/Implementation/PerformanceCounterData.cs

```diff
@@ -17,11 +17,27 @@ internal class PerformanceCounterData
             this.UsesInstanceNamePlaceholder = usesInstanceNamePlaceholder;
             this.IsCustomCounter = isCustomCounter;
             this.IsInBadState = isInBadState;
```

Comment:
What is "IsInBadState" used for?  And how do you know if that is true or false when you create an instance of this class?